test-validator/src/lib.rs

```diff
@@ -1146,22 +1146,46 @@ impl Drop for TestValidator {
 mod test {
     use super::*;
 
+    impl TestValidator {
+        // Wait for the given slot to be observed as confirmed. Returns true if
+        // the slot (or a greater one) is observed as confirmed, false otherwise.
+        fn wait_for_confirmed_slot(&self, slot: Slot) -> bool {
```

Comment:
Ugh sorry, thought I had gotten all tests fixed up but I see another one failed in CI. The way health check is implemented inherently adds a race condition for test validator as send_transaction() does a health check. I didn't observed the race causing a failure locally but they are clearly there as CI is hitting them.

I added a workaround for this one test, but I'm thinking that this will be a nuisance / spot for potential future race conditions.

I'm currently leaning towards just setting the override for health in TestValidator such that it always returns healthy. There is no sense of a cluster with a test validator, so we shouldn't bother checking it. Additionally, this is basically equivalent to behavior before this PR as the test validator would not have had any known validators and would just return `Ok` every time because of that.

Reply:
> I'm currently leaning towards just setting the override for health in TestValidator such that it always returns healthy.

I think that makes sense.

Reply:
Sweet, i"m going to go that route. And after not working on this for half a day and taking a step back, the following bit makes this an obvious decision in my mind:
> Additionally, this is basically equivalent to behavior before this PR as the test validator would not have had any known validators and would just return Ok every time because of that.